homeassistant/components/onewire/switch.py

```diff
@@ -0,0 +1,204 @@
+            "path": "latch.0",
+            "name": "Latch 0",
+            "type": SWITCH_TYPE_LATCH,
+            "default_disabled": True,
+        },
+        {
+            "path": "latch.1",
+            "name": "Latch 1",
+            "type": SWITCH_TYPE_LATCH,
+            "default_disabled": True,
+        },
+        {
+            "path": "latch.2",
+            "name": "Latch 2",
+            "type": SWITCH_TYPE_LATCH,
+            "default_disabled": True,
+        },
+        {
+            "path": "latch.3",
+            "name": "Latch 3",
+            "type": SWITCH_TYPE_LATCH,
+            "default_disabled": True,
+        },
+        {
+            "path": "latch.4",
+            "name": "Latch 4",
+            "type": SWITCH_TYPE_LATCH,
+            "default_disabled": True,
+        },
+        {
+            "path": "latch.5",
+            "name": "Latch 5",
+            "type": SWITCH_TYPE_LATCH,
+            "default_disabled": True,
+        },
+        {
+            "path": "latch.6",
+            "name": "Latch 6",
+            "type": SWITCH_TYPE_LATCH,
+            "default_disabled": True,
+        },
+        {
+            "path": "latch.7",
+            "name": "Latch 7",
+            "type": SWITCH_TYPE_LATCH,
+            "default_disabled": True,
+        },
+    ],
+}
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up 1-Wire platform."""
+    # Only OWServer implementation works with switches
+    if config_entry.data[CONF_TYPE] == CONF_TYPE_OWSERVER:
+        onewirehub = hass.data[DOMAIN][config_entry.unique_id]
+
+        entities = await hass.async_add_executor_job(get_entities, onewirehub)
+        async_add_entities(entities, True)
+
+
+def get_entities(onewirehub: OneWireHub):
+    """Get a list of entities."""
+    entities = []
+
+    for device in onewirehub.devices:
+        family = device["family"]
+        device_type = device["type"]
+        sensor_id = os.path.split(os.path.split(device["path"])[0])[1]
+
+        if family not in DEVICE_SWITCHES:
+            continue
+
+        device_info = {
+            "identifiers": {(DOMAIN, sensor_id)},
+            "manufacturer": "Maxim Integrated",
+            "model": device_type,
+            "name": sensor_id,
+        }
+        for device_switch in DEVICE_SWITCHES[family]:
+            device_file = os.path.join(
+                os.path.split(device["path"])[0], device_switch["path"]
+            )
+            entities.append(
+                OneWireSwitch(
+                    sensor_id,
+                    device_file,
+                    device_switch["type"],
+                    device_switch["name"],
+                    device_info,
+                    device_switch.get("default_disabled", False),
+                    onewirehub.owproxy,
+                )
+            )
+
+    return entities
+
+
+class OneWireSwitch(SwitchEntity, OneWireProxy):
```

Comment:
It's safer to invert the inheritence order and remove the non common properties from the custom integration parent entity class `OneWireProxy`. Ie remove the `state` property from that class and let `state` be implemented by each platform.

Reply:
New PR created to address this comment: https://github.com/home-assistant/core/pull/43019